### ISSUE TITLE
[HOLD for payment 2025-02-05] [$250] Workspace - User can navigate to deleted workspace editor while offline

### ISSUE THREAD

**Issue by vincdargento** [Expensify/App#54582]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.78-4
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** N/A
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Create a workspace
2. Navigate to the workspace and enable workflows
3. Go to connect bank account and copy the URL
4. Go offline
5. Delete the workspace
6. Send the copied URL in any chat
7. Click on it to navigate to the editor
8. Change any value while offline

## Expected Result:
User should not be able to navigate to workspace editor after the workspace has been deleted (as in PR #27743)

## Actual Result:
User is able to navigate to workspace editor in a deleted workspace while offline

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/ba34b5b1-4334-4286-935c-0dc480049dd7

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021873785256425821183</li>
        <li>Upwork Job ID: 1873785256425821183</li>
        <li>Last Price Increase: 2025-01-13</li>
<li>Automatic offers: </li>
<ul>
<li>rayane-djouah | Reviewer | 105681465</li>
<li>FitseTLT | Contributor | 105681467</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @slafortune</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
User is able to navigate to workspace editor in a deleted workspace while offline

### What is the root cause of that problem?
The condition to display not found page now doesn't check the policy is pending delete 

https://github.com/Expensify/App/blob/56e35ce5f0d88551149f1b21b7a8f39ee42a0bf3/src/pages/ReimbursementAccount/ReimbursementAccountPage.tsx#L436-L439
### What changes do you think we should make in order to solve the problem?
We should display not found page if the policy is pending delete

```
if (!isLoading && (isEmptyObject(policy) || !PolicyUtils.isPolicyAdmin(policy) || PolicyUtils.isPendingDeletePolicy(policy))) {
```

https://github.com/Expensify/App/blob/56e35ce5f0d88551149f1b21b7a8f39ee42a0bf3/src/pages/ReimbursementAccount/ReimbursementAccountPage.tsx#L436-L439

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)
NA

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by nkdengineer** [comment]
Updated [proposal](https://github.com/Expensify/App/issues/54582#issuecomment-2562987255) to fix other places if we want to fix this here.

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace - User can navigate to deleted workspace editor while offline

### What is the root cause of that problem?
We already show not found page for pending delete for both WorkspaceInitialPage 
https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspaceInitialPage.tsx#L351
and WorkspacePageWithSections
https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspacePageWithSections.tsx#L163
but that only happens in online mode b/c shouldShowPolicy returns true for pending delete workspaces for offline case
https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/libs/PolicyUtils.ts#L218-L223
this behaviour is correct for WorkspaceListPage [here](https://github.com/FitseTLT/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspacesListPage.tsx#L343) because we want to show the pending delete workspace in offline mode but not in these places.
### What changes do you think we should make in order to solve the problem?

We should first rename isOffline param of shouldShowPolicy to `shouldShowPendingDeleteWorkspace` then we pass false for the shouldShowPolicy we are using in
https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspacePageWithSections.tsx#L154-L155
and additionally here to display not found for the case of worspace inital setting page 
https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspaceInitialPage.tsx#L347-L348
And additionally for WorkspacePageWithSections there is a problem of offline indicator showing on the top of the page for small screen (because we shouldForceFullScreen that will make the view fixed position and it distorts the normal display of the offline indicator on the bottom of the screen)  so as we don't show the wide screen offline indicator when not found page is displayed [here](https://github.com/FitseTLT/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspacePageWithSections.tsx#L173) we can make it consistent with it and disable the small screen offline indicator when displaying not found page so add
```
            shouldShowOfflineIndicator={!shouldShow}

```
but if we want to show offline indicator when displaying not found page in WorkspacePageWithSections like we do in WorkspaceInitialPage we can make the shouldForceFullScreen [here](https://github.com/FitseTLT/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspacePageWithSections.tsx#L180) to apply only to large screens (!shouldUseNarrowLayout or !isSmallScreenWidth) because we don't need it for small screen and the offline indicator will display in the bottom correctly.
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by FitseTLT** [comment]
@rayane-djouah Yep that's what I suggested in my proposal we can indeed use AccessOrNotFoundWrapper with the details to be thought on PR phase. And isPolicyNotAccessible condition as you suggested in AccessOrNotFoundWrapper after updating shouldShowPolicy to exclude pending deleted by using it's second param just I have detailed in my [proposal](https://github.com/Expensify/App/issues/54582#issuecomment-2563098388) but we can't updated shouldShowPolicy to always avoid pending deleted policies because we want to show pending delete on workspace list page.

shouldShowPolicy: we should use it to determine if the workspace can be shown in the workspace list page so we show pending deleted policies in offline mode
isPolicyAccessible: we are using it to check the policy is not empty has proper id and keys and no errors so we can use incorporate the new logic of disallowing pending delete policies

**Comment by FitseTLT** [comment]
@rayane-djouah Yep that's what I suggested in my proposal we can indeed use AccessOrNotFoundWrapper with the details to be thought on PR phase.

shouldShowPolicy: we should use it to determine if the workspace can be shown in the workspace list page so we show pending deleted policies in offline mode and if the policy has errors or If it is join request pending and so on.

isPolicyAccessible: we are using it to check the policy is not empty has proper id and keys and no errors so we can incorporate the new logic of disallowing pending delete policies in it. We use it in AccessOrNotFoundWrapper and to determine if we should push the workspace initial page on the nav state here
https://github.com/Expensify/App/blob/7ea58c74906bf0dcf2b65a10603b9decb7c7fc65/src/libs/Navigation/AppNavigator/createCustomFullScreenNavigator/CustomFullScreenRouter.tsx#L33-L36

So they have different purposes and regarding not found page for non-accessible policies isPolicyAccessible is the most appropriate function so we can update isPolicyAccessible logic to exclude pending delete workspaces and use it in AccessOrNotFoundWrapper to cover a lot of workspace pages and also in [WorkspacePageWithSections.tsx](https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspacePageWithSections.tsx#L154-L155), [WorkspaceInitialPage.tsx](https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/workspace/WorkspaceInitialPage.tsx#L347-L348) and other pages. But we work on the detail in the PR phase 👍

**Comment by amyevans** [comment]
@FitseTLT how's the PR coming along?

**Comment by slafortune** [comment]
@rayane-d can you please accept the offer - https://www.upwork.com/nx/wm/offer/105681465

**Comment by slafortune** [comment]
OH! I looked at those dates backwards!  Thanks for calling it out.

**Comment by slafortune** [comment]
@FitseTLT Role Contributor - was paid $250 via Upwork
@rayane-djouah Role C+ - requires payment via NewDot - owed $250

**Comment by slafortune** [comment]
Upworks offer withdrawn
